arkouda/strings.py

```diff
@@ -596,6 +592,12 @@ def _get_matcher(self, pattern: Union[bytes, str_scalars], create: bool = True):
             re.compile(pattern)
         except Exception as e:
             raise ValueError(e)
+        if pattern == "$" or (pattern == "" and (self == "").any()):  # type: ignore
+            # TODO remove once changes from chapel issue #20431 and #20441 are in arkouda
+            raise ValueError(
+                "regex operations not currently supported with a pattern=$ or pattern='' when "
+                "the empty string is contained in Strings"
+            )
```

Comment:
Can we pull this out into a subfunction since we repeat the same code for this in 3 places?